packages/eslint-plugin/configs/recommended-with-formatting.js

```diff
@@ -40,4 +45,23 @@ const config = {
 	},
 };
 
+// Don't apply Jest config if Playwright is installed.
+if (
+	isPackageInstalled( 'jest' ) &&
+	! isPackageInstalled( '@playwright/test' )
+) {
+	config.overrides = [
+		{
+			// Unit test files and their helpers only.
+			files: [ '**/@(test|__tests__)/**/*.js', '**/?(*.)test.js' ],
```

Comment:
Interesting though, I found out that there might be a bug in the older versions (`v3.0.4`) of `minimatch`, which causes the inconsistent results:

```js
minimatch_3_0_4.makeRe(`**/@(test|__tests__)/**/*.js`).test('packages/warning/test/babel-plugin.js') // false
minimatch_5_0_1.makeRe(`**/@(test|__tests__)/**/*.js`).test('packages/warning/test/babel-plugin.js') // true
```

We're using `minimatch@3.0.4` in Gutenberg, which means the overrides here won't match some of the test files (`packages/warning/test/babel-plugin.js` for instance). Upgrading minimatch or eslint should fix the issue.

However, `**/test/**/*.js` still works as expected, and I added it in the root `.eslintrc.js` in this PR, so these files will start to get picked up from now on.

Reply:
It would be great to update `minimatch` to the latest version in a separate PR. I think we might have similar issues in the past with one of the glob libraries (which still might depend on `minimatch`).